### DISCUSSION TITLE
ZepetoPlayers.instance.controllerData.horizontalController.SetActive(false);

### DISCUSSION THREAD

**Discussion by smilejsu82** [naverz/zepeto-studio-kor#675] [5. Q&A]
월드 1.2.0입니다 

처음 시작할때 UI를 안보여주고 준비가 되면 UI를 보여주려고 합니다 
`ZepetoPlayers.instance.controllerData.horizontalController.SetActive(false);`
를 사용했다가 
`ZepetoPlayers.instance.controllerData.horizontalController.SetActive(true);`

요롷게 하려고 했는데 동작을 안하더라구요 

그래서 일단 임시로 
`private* WaitForLoadUIPlayerControl() {
        console.log(
            "---------------- WaitForLoadUIPlayerControl ------------------"
        );

        while (true) {
            this.uiControl = GameObject.FindObjectOfType<UIZepetoPlayerControl>();
            console.log(`========> uiControl : ${this.uiControl}`);
            if (this.uiControl) break;
            yield null;
        }

        this.uiControl.gameObject.SetActive(false);
    }`

요롷게 만들어서 사용중인데요 

`ZepetoPlayers.instance.controllerData.horizontalController.SetActive(true);``

이부분 동작 하는건가요? 

이상하게 
`ZepetoPlayers.instance.controllerData.horizontalController.SetActive(false);``는 또 동작 하드라구요 ~.~

https://user-images.githubusercontent.com/50188795/165713178-3683879c-cbb0-43bf-9b22-58c2c2365527.mp4

**Comment by pumfit** [comment]
기존 컨트롤러를 해제하고 직접 컨트롤러를 만들어 제어하는 방식을 사용하시면 좋을 것 같습니다!
ZepetoPlayers에서 Use Default Virtual Controller 옵션을 꺼주고
<img width="346" alt="image" src="https://user-images.githubusercontent.com/46295539/165880290-9073ebe6-dcb8-4526-99e0-bbba796fc837.png">
아래 코드와 같이 컨트롤러를 직접 생성하며 원하는 타이밍에 컨트롤러를 SetActive할 수 있습니다!
저는 예시로 버튼을 사용해 보았는데 잘 작동됩니다~!
<img width="736" alt="image" src="https://user-images.githubusercontent.com/46295539/165880500-bd000865-6e1b-4a2a-971d-6929fdaf245b.png">

**Reply by smilejsu82** [reply]
오옷 이런 방법이 있었군요 ~ 감사합니다 
일단은 지금 잘 돌아가고 있어서 나중에 수정 할때 바꿔 봐야 겠어요 감사합니다 ~ (__)
